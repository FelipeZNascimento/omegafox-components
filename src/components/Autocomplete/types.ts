export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
