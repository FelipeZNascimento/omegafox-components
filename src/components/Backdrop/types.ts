export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
