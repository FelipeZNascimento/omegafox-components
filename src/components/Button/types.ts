export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
