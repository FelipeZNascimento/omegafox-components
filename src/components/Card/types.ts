export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
