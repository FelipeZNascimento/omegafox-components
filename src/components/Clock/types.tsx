import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
