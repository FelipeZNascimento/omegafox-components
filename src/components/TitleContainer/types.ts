import { TBorderPosition } from 'components/types';

export interface ITitleContainerProps {
  borderPosition?: TBorderPosition;
  size?: 'small' | 'big';
  text: string;
}
