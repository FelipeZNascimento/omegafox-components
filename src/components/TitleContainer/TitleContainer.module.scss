@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.titleContainer {
  @include flex-aligned(center, center);
  @include bg-colored($color-qatar-red);
  @include font-colored($color-grey1);

  -webkit-box-shadow: 0px 0px 0px 2px $color-grey3;
  box-shadow: 0px 0px 0px 2px $color-grey3;

  border-radius: $border-radius;
  display: inline-flex;

  &Small {
    @include font-sized(l);
    @include padded(all, m);

    min-height: calc($title-container-height / 2);
  }
  &Big {
    @include font-sized(l3);
    @include padded(all, l);

    min-height: $title-container-height;
  }

  &BottomLeft {
    border-bottom-left-radius: 0;
  }
  &BottomRight {
    border-bottom-right-radius: 0;
  }
  &TopRight {
    border-top-right-radius: 0;
  }
  &TopLeft {
    border-top-left-radius: 0;
  }
}
