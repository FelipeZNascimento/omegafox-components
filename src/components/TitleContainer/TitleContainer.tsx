import React from 'react';

import classNames from 'classnames';
import { ITitleContainerProps } from './types';
import styles from './TitleContainer.module.scss';

export const TitleContainer = ({
  borderPosition = 'bottomLeft',
  size = 'big',
  text
}: ITitleContainerProps) => {
  const containerClass = classNames(styles.titleContainer, {
    [styles.titleContainerBottomLeft]: borderPosition === 'bottomLeft',
    [styles.titleContainerBottomRight]: borderPosition === 'bottomRight',
    [styles.titleContainerTopLeft]: borderPosition === 'topLeft',
    [styles.titleContainerTopRight]: borderPosition === 'topRight',
    [styles.titleContainerSmall]: size === 'small',
    [styles.titleContainerBig]: size === 'big'
  });

  return <div className={containerClass}>{text}</div>;
};
