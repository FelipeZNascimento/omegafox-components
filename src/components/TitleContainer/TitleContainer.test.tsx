import React from 'react';
import { render } from '@testing-library/react';
import { TitleContainer } from './TitleContainer';
import { ITitleContainerProps } from './types';

describe('<TitleContainer />', () => {
  const renderComponent = ({ borderPosition, text }: ITitleContainerProps) =>
    render(<TitleContainer borderPosition={borderPosition} text={text} />);

  it('should render', () => {
    const text = 'Click';
    const { container } = renderComponent({ text: text });

    expect(container).toMatchSnapshot();
  });
});
