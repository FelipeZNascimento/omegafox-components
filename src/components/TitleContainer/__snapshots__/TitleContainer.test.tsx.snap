// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleContainer /> should render 1`] = `
<div>
  <div
    class="titleContainer titleContainerBottomLeft titleContainerBig"
  >
    Click
  </div>
</div>
`;
